### PR TITLE
fix panic caused by series endpoint with remote_read

### PR DESCRIPTION
**Pull request by y2sarakawa**
This is for issue #4240.

PR #4213 was complicated, so this is its rewrite based on release-2.3.

### CODE REVIEW

File: web/api/v1/api_test.go [hunk 1]
```diff
@@ -646,6 +649,90 @@ func TestReadEndpoint(t *testing.T) {
...
+	}
+	seriesURL.RawQuery = q.Encode()
+	req, err := http.NewRequest("GET", seriesURL.String(), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, apiErr, _ := api.series(req)
+	if apiErr != nil {
+		if apiErr.typ == errorExec {
```
Comment (henridf) [inline web/api/v1/api_test.go]: I realize you had a request on another PR to test the series/ endpoint, but having to whitelist various connection errors this way seems fragile.... I'd say that just testing the changed function (`ToQuery`) in `codec_test.go` would be ok.

Comment (y2sarakawa) [reply web/api/v1/api_test.go]: Thanks. I will cancel this PR.

**Comment by y2sarakawa** [comment]
I will recreate another PR that excluded the unit test later.

**Comment by henridf** [comment]
hi @y2sarakawa i'm happy to open a PR with your change and push it through if you'd like. just ping me here if so.

**Comment by y2sarakawa** [comment]
#4263 was opened. I have nothing to do.

**Pull request closed without merging**